incubator/exporter-otlp.spec/src/main/scripts/io/aklivity/zilla/specs/exporter/otlp/schema/otlp.schema.patch.json

```diff
@@ -0,0 +1,66 @@
+[
+    {
+        "op": "add",
+        "path": "/$defs/telemetry/exporter/properties/type/enum/-",
+        "value": "otlp"
+    },
+    {
+        "op": "add",
+        "path": "/$defs/telemetry/exporter/allOf/-",
+        "value":
+        {
+            "if":
+            {
+               "properties":
+                {
+                    "type":
+                    {
+                        "const": "otlp"
+                    }
+                }
+            },
+            "then":
+            {
+                "properties":
+                {
+                    "type":
+                    {
+                        "const": "otlp"
+                    },
+                    "options":
+                    {
+                        "properties":
+                        {
+                            "interval":
+                            {
+                                "type": "number"
+                            },
+                            "endpoints":
+                            {
+                                "type": "array",
+                                "items":
+                                {
+                                    "properties":
+                                    {
+                                        "url":
```

Comment:
I modeled this after the prometheus exporter. I don't think the object is really necessary, we can get rid of that.

As for the multiple endpoints, it was your request at the prometheus exporter, so I did the same thing here for consistency. However, I am not sure what the reasoning was in that case and if it applies here, too. 🤔

So, I believe the config can be simplified from:
```
      type: otlp
      options:
        interval: 5
        endpoints:
          - url: http://localhost:4318/v1/metrics
```
to:
```
      type: otlp
      options:
        interval: 5
        endpoint: http://localhost:4318/v1/metrics # pls note: endpoints ➡️ endpoint
```

Then it will look a bit different from the prometheus one, though, which currently looks like this:
```
      type: prometheus
      options:
        endpoints:
          - scheme: http
            path: /metrics
            port: 9999
```

We need to make a decision here about the config format.

Reply:
There's no problem if different exporters have different syntax, they should be adopting the most natural syntax for their specific behavior. The same is true for bindings, vaults and guards too.

For `otlp` exporter, seems like we would want to be able to configure options that handle configuration similar to the Java agent, please see https://opentelemetry.io/docs/concepts/sdk-configuration/otlp-exporter-configuration/.

Reply:
how about this format?

```
      type: otlp
      options:
        interval: 5
        endpoint:
          http:
            metrics: http://localhost:4318/v1/metrics
            # we can add these in the future:
            #logs:
            #traces:
          #grpc:
            #metrics:
            #logs:
            #traces:
```

Reply:
FTR we decided on this format:

without overrides
```
type: otlp
      options:
        interval: 5
        signals: # default: expose all signals
          - metrics
          - logs
          - traces
        endpoint:
          #protocol: http # http should be the default or grpc
          location: http://localhost:4318/ # this sets the default for metrics, logs, traces e.g. http://localhost:4318/v1/metrics
```
with overrides
```
      type: otlp
      options:
        interval: 5
        signals:
          - metrics
          - logs
          - traces
        endpoint:
          #protocol: http # default
          location: http://localhost:4318/
          overrides:
            metrics: http://localhost:4318/v1/customUrlForMetrics # or metrics: /v1/customPathForMetrics
            #logs:
            #traces:
```